libs/auth-api-lib/src/lib/delegations/delegations-index.service.ts

```diff
@@ -342,57 +397,152 @@ export class DelegationsIndexService {
-          toNationalId: user.nationalId,
-          type: AuthDelegationType.LegalGuardian,
-          provider: AuthDelegationProvider.NationalRegistry,
+          fromNationalId,
+          type: AuthDelegationType.Custom,
+          provider: AuthDelegationProvider.Custom,
+          customDelegationScopes: { [Op.contains]: [scope.name] },
+          validTo: { [Op.or]: [{ [Op.gte]: new Date() }, { [Op.is]: null }] },
         },
-      },
-    )
+      })
+      .then((d) => d.map((d) => d.toDTO()))
+  }
+
+  private async getRepresentativeDelegationRecords({
+    scope,
+    fromNationalId,
+  }: FetchDelegationRecordsArgs): Promise<DelegationRecordDTO[]> {
+    if (!scope.grantToPersonalRepresentatives) {
+      return []
+    }
+
+    // Get all personal representative right types that are permitted for the scope and construct the delegation types
+    const permittedDelegationTypes =
+      await this.personalRepresentativeScopePermissionService
+        .getScopePermissionsAsync(scope.name)
+        .then((scopePermission) =>
+          scopePermission.map((rightType) =>
+            getPersonalRepresentativeDelegationType(rightType.rightTypeCode),
+          ),
+        )
+
+    return this.delegationIndexModel
+      .findAll({
+        where: {
+          fromNationalId,
+          type: {
+            [Op.in]: permittedDelegationTypes,
+          },
+          provider: AuthDelegationProvider.PersonalRepresentativeRegistry,
+          validTo: { [Op.or]: [{ [Op.gte]: new Date() }, { [Op.is]: null }] },
+        },
+      })
+      .then((d) => d.map((d) => d.toDTO()))
+  }
+
+  private async getCompanyDelegationRecords({
+    scope,
+    fromNationalId,
+  }: FetchDelegationRecordsArgs): Promise<DelegationRecordDTO[]> {
+    if (!scope.grantToProcuringHolders) {
+      return []
+    }
 
-    return this.sortDelegation(currentDelegationIndexItems, delegations)
+    return this.delegationIndexModel
+      .findAll({
+        where: {
+          fromNationalId,
+          type: {
+            [Op.in]:
+              delegationProviderTypeMap[AuthDelegationProvider.CompanyRegistry],
+          },
+          provider: AuthDelegationProvider.CompanyRegistry,
+          validTo: { [Op.or]: [{ [Op.gte]: new Date() }, { [Op.is]: null }] },
+        },
+      })
+      .then((d) => d.map((d) => d.toDTO()))
+  }
+
+  private async getWardDelegationRecords({
+    scope,
+    fromNationalId,
+  }: FetchDelegationRecordsArgs): Promise<DelegationRecordDTO[]> {
+    if (!scope.grantToLegalGuardians) {
+      return []
+    }
+
+    return this.delegationIndexModel
+      .findAll({
+        where: {
+          fromNationalId,
+          type: {
+            [Op.in]:
+              delegationProviderTypeMap[
+                AuthDelegationProvider.NationalRegistry
+              ],
+          },
+          provider: AuthDelegationProvider.NationalRegistry,
+          validTo: { [Op.or]: [{ [Op.gte]: new Date() }, { [Op.is]: null }] },
```

Comment:
We don't get `validTo` from National Registry, but I started thinking about if we should actually populate this based on the date when the child becomes 18 year old, to prevent edge cases on indexing not been triggered after the child turned 18?

Reply:
Done ✅